media/js/app/taxonomy/taxonomy.js

```diff
@@ -486,110 +488,42 @@
-                                                ) {
-                                                    var tempT = new Term({
-                                                        'display_name': parents[
-                                                            parents.indexOf(
-                                                                self.findUtil(
-                                                                    parents, it.attributes.display_name)[0])]
-                                                            .term_set[z].display_name,
-                                                        'vocabulary_id': it.attributes.id
-                                                    });
-                                                    tempT.save({}, {
-                                                        success: function(itT) {
-                                                            parents[
-                                                                parents.indexOf(
-                                                                    self.findUtil(
-                                                                        parents, it.attributes.display_name)[0])]
-                                                                .self.get('term_set').add(itT);
-                                                            self.render();
-                                                        }
-                                                    });
-                                                }
-                                            }
-                                        });
-                                    } else {
-                                        for (var z = 0; z < parents[parents.indexOf(self.findUtil(parents, model_search.attributes.display_name)[0])].term_set.length; z++) {
-                                            var tempT = new Term({
-                                                'display_name': parents[parents.indexOf(self.findUtil(parents, model_search.attributes.display_name)[0])].term_set[z].display_name,
-                                                'vocabulary_id': model_search.attributes.id
-                                            });
-                                            tempT.save({}, {
-                                                success: function(itT) {
-                                                    model_search.get('term_set').add(itT);
-                                                    self.render();
-                                                }
-                                            });
-                                        }
-                                    }
-                                }(j));
-                            }
-                        }
                     } else if (display !== undefined && display.length > 0) {
                         var urls = selectedVocabulary.getOnomyUrls();
```

Comment:
Could you please elaborate on how you think it could be saved multiple times? My intention is to add the onomy_url to the vocabulary if it doesn't already contain it then, if the term doesn't exist add it to the vocabulary.